src/main/scala/com/microsoft/hyperspace/actions/RefreshAppendAction.scala

```diff
@@ -108,7 +88,10 @@ class RefreshAppendAction(
     //   https://github.com/microsoft/hyperspace/issues/183
 
     // New entry.
-    entry.copy(content = mergedContent)
+    entry
+      .copy(content = mergedContent)
+      .withNewDeletedFiles(deletedFiles)
```

Comment:
It seems we shouldn't allow deletedFiles in `RefreshAppendAction` to handle modified files properly. Please check the below scenario I commented before:

==
Also I think we should consider the following scenario:

- index creation without linage column
- some of source files are modified
- refresh incremental mode:
  - delete => failed, do nothing
  - append => should throw an exception

I think this can be ensured by adding `deletedFiles.isEmpty` condition in `validate` of `RefreshAppendAction`.

Reply:
> refresh incremental mode:
> delete => failed => 

thanks @sezruby , I guess this comment is still applicable only for this PR #187 and not for the current PR, right?
In any case, if delete fails, the application will throw an exception, it will not even go to the next step. `RefreshAppendAction` will not be called at all. So I don't think this assert is necessary in `validate`.

> append => should throw an exception

We never reach this step, previous step will throw an exception before coming here.

Now if you are talking about independently calling `RefreshAppendAction` outside of `refreshIndex(mode = "incremental")`, I still don't think we should add `deleted` check in append validation. It's because in future, we will have `refeshIncremental(mode = "quick")` which will populate the "deleted" in metadata without modifying the index.
If we call `RefreshAppendAction` independently, it should not `throw` if "deleted" is non-empty. It should just update the "deleted" list with additional deleted files and move on.

If you look at `def logEntry`, the "unhandled" deleted files are being added to the "deleted" list at the end of the operation. Does that solve the issue?

Reply:
In my opinion, `quick` mode should be handled with a different action. Yes in general the condition in `validate` is not necessary. But just for double check in case concurrently deleted/modified files and I don't want to let unhandled `deleted files` after `RefreshAppendAction` which may cause some confusion later. How do we handle if there's `deletedFiles` in the latest index log entry? in 0.4 ver? 

BTW, what if a file is modified/appended/deleted in the middle of refresh? Though we don't need to be completely transactional, I think at least we should use the same source file list at the specific point in time and consistent with signature calculation, source content capture and processing appended, deleted files?
WDYT? @apoorvedave1 
cc @imback82

Reply:
Thanks @sezruby , @imback82 . Closing this after offline discussions. As of now, the consensus is on the current behavior of the PR.